OEP-6/OEP-6.mediawiki

```diff
@@ -0,0 +1,452 @@
+
+* Rejects with '''MALFORMED_PUBLIC_KEY''' in case the <code>public key</code> is not hex encoded
+* Rejects with '''MALFORMED_MESSAGE''' in case the <code>message</code> is not hex encoded
+* Rejects with '''MALFORMED_SIGNATURE''' in case the <code>signature</code> is not hex encoded
+
+
+====signMessage====
+<pre>
+function signMessage(address: string, message: string): Promise<string>
+</pre>
+
+Initiates signing of text <code>message</code> by the account or identity. The <code>address</code> '''must''' be one of the accounts returned by <code>getOwnAccounts</code> or identities returned by <code>getOwnIdentities</code>.
+
+This method provide the dAPI provider with the message body, which it should display to the user prior to signing (e.g.: a contract to sign).
+
+* Rejects with '''NO_ADDRESS''' in case the user is not signed in or has no account or identity
+* Rejects with '''WRONG_ADDRESS''' in case the <code>address</code> is not one of own accounts or identites
+* Rejects with '''MALFORMED_ADDRESS''' in case the <code>address</code> is not a valid account or identity
+
+
+====verifyMessage====
+<pre>
+function verifyMessage(address: string, message: string, address: string, signature: string): Promise<boolean>
+</pre>
+
+Verifies that the <code>signature</code> is created by the account or identity over text <code>message</code>. The <code>address</code> '''must''' be one of the accounts returned by <code>getOwnAccounts</code> or identities returned by <code>getOwnIdentities</code>.
+
+* Rejects with '''NO_ADDRESS''' in case the user is not signed in or has no account or identity
+* Rejects with '''WRONG_ADDRESS''' in case the <code>address</code> is not one of own accounts or identites
+* Rejects with '''MALFORMED_ADDRESS''' in case the <code>address</code> is not a valid account or identity
+* Rejects with '''MALFORMED_MESSAGE''' in case the <code>message</code> is not hex encoded
+
+
+====verifyMessagePk====
+<pre>
+function verifyMessage(publicKey: string, message: string, address: string, signature: string): Promise<boolean>
+</pre>
+
+Verifies that the <code>signature</code> is created by a private key corresponding to <code>publicKey</code> over text <code>message</code>. 
+
+* Rejects with '''MALFORMED_PUBLIC_KEY''' in case the <code>public key</code> is not hex encoded
+* Rejects with '''MALFORMED_SIGNATURE''' in case the <code>signature</code> is not hex encoded
+
+
+===SmartContract===
+The main functionality of this component is invocation of smart contract methods.
+
+====invoke====
+<pre>
+type ParameterType = 'Boolean' | 'Integer' | 'ByteArray' | 'Struct' | 'Map' | 'String';
+
+interface Parameter {
+    type: ParameterType;
+    value: any;
+}
+
+function invoke(account: string, contract: string, method: string, parameters: Parameter[], 
```

Comment:
Account should be address.

The gaslimit and gasprice are tricky. Every instruction in SC call costs something. So only the dApp developer know how costly is the call. On the other hand we have the gasprice value which user can increase in times when the network is congested. So his transaction get processed sooner.

Therefore we should let user decide what unit price he is willing to pay, but let the dApp developer set the limit. But because he does not know what unit price the user will choose, it should be stated relative to the unit price.

And after all, user is gonna pay for the transaction, so he should have the option to override the limit. DApp developer should only tell: "this operation is costly, you should use this limit"